### PR TITLE
implement handleRevert

### LINKED ISSUE #5069: Implement getRevertReason for 4.x
implement all things related to `handleRevert:true` setting

Documentation:
https://web3js.readthedocs.io/en/v1.7.3/web3-eth.html?highlight=handlerevert#handlerevert
https://web3js.readthedocs.io/en/v1.7.3/web3-eth-contract.html?highlight=handlerevert#handlerevert

Implementation in 1.x
https://github.com/ChainSafe/web3.js/blob/2c0324af2da467ee1acb72452f20000e916e4306/packages/web3-core-method/src/index.js#L917

https://github.com/ChainSafe/web3.js/blob/2c0324af2da467ee1acb72452f20000e916e4306/packages/web3-core-method/src/index.js#L442

Also, add integration test in `web3-eth/test/integration/defaults.test.ts`

### PR DESCRIPTION
**Pull request by nikoulai**
## Description

#5069

### CODE REVIEW

File: packages/web3-eth/src/rpc_method_wrappers.ts [hunk 1]
```diff
@@ -1071,6 +1071,14 @@ export function sendTransaction<
 							ETH_DATA_FORMAT,
 						);
 
+						if (web3Context.handleRevert) {
+							// eslint-disable-next-line no-use-before-define
+							await getRevertReason(
+								web3Context,
+								transactionFormatted as TransactionCall,
+							);
```
Comment (nazarhussain) [inline packages/web3-eth/src/rpc_method_wrappers.ts]: The return format should also pass to this function.

File: packages/web3/test/shared_fixtures/build/Basic.json [hunk 2]
```diff
@@ -0,0 +1,253 @@
+{
```
Comment (nazarhussain) [inline packages/web3/test/shared_fixtures/build/Basic.json]: Would be preferred to create a symlink to the `build` folder or particular file.

**Review by nazarhussain — CHANGES_REQUESTED** [review]

File: packages/web3/test/shared_fixtures/build/Basic.json (hunk 2 above)
Comment (avkos) [reply packages/web3/test/shared_fixtures/build/Basic.json]: I think, parent folder `shared_fixtures` should be a symlink

**Review by avkos — CHANGES_REQUESTED** [review]

File: packages/web3/test/shared_fixtures/build/Basic.json (hunk 2 above)
Comment (nikoulai) [reply packages/web3/test/shared_fixtures/build/Basic.json]: we have `system_tests_utils.ts` in `shared_fixtures`, which is not in the original folder (@avkos )

File: packages/web3-eth/src/rpc_method_wrappers.ts [hunk 3]
```diff
@@ -1071,6 +1071,15 @@ export function sendTransaction<
 							ETH_DATA_FORMAT,
 						);
 
+						if (web3Context.handleRevert) {
```
Comment (jdevcs) [inline packages/web3-eth/src/rpc_method_wrappers.ts:1074]: LGTM, just mention in doc that `handleRevert` is available for `sendTransaction` only for now.

**Review by avkos — APPROVED** [review]

**Review by nazarhussain — APPROVED** [review]

**Pull request merged**